cypress/integration/database-page.spec.js

```diff
@@ -0,0 +1,181 @@
+
+        const assetFiltersHaveChanged = () => {
+            getReasearchDatabases()
+                .then(filteredDatabases => {
+                    numberOfFilteredDatabases = filteredDatabases.length;
+                    expect(filteredDatabases.length).to.be.below(
+                        numberOfResearchDatabases
+                    );
+                });
+        };
+
+        const getActiveTags = () => cy.get('.tag-list').find('.active');
+
+        const getClearButton = () =>
+            cy
+                .get('button')
+                .contains('Clear filters');
+
+        const getFilterButton = (familyFilterButton, targetLabelText) => 
+            familyFilterButton
+                .closest('.filter-family')
+                .find('ul > li label')
+                .contains(targetLabelText)
+                .then(updateActiveFilters);
+
+        const getFirstFilterFamilyButton = () => 
+            cy
+                .wait(500)
+                .get('@filterButtons')
+                .first();
+        
+        const updateActiveFilters = filterButton => {
+            activeFilters.push(
+                filterButton && filterButton.length
+                    ? filterButton[0].innerText.trim()
+                    : ''
+            );
+        };
+
+        /** End of Helper Functions */
+
+        let numberOfFilteredDatabases = 0;
+        let activeFilters = [];
+
+        beforeEach(() => {
+            cy
+                .get(constants.selectors.filters)
+                .find(constants.selectors.filterButton)
+                .as('filterButtons');
+        });
+
+        it('should filter records based on the first filter', () => {
+            const filterFamilyButton = getFirstFilterFamilyButton()
+                .click()
+                .wait(1000);
+
+            getFilterButton(filterFamilyButton, 'Biography')
+                .click();
+
+            assetFiltersHaveChanged();
+        });
+
+        it('should show cards with the selected filter', () => {
+            getActiveTags().each(assertActiveFilterMatch);
+        });
+
+        it('should allow for multiple filters to be applied', () => {
+            const filterFamilyButton = getFirstFilterFamilyButton()
+                .wait(1000);
+
+            getFilterButton(filterFamilyButton, 'History')
+                .click();
+
+            assetFiltersHaveChanged();
+        });
+
+        it('should show cards with the selected multiple filter', () => {
+            getActiveTags().each(assertActiveFilterMatch);
+        });
+
+        it('should be able to remove a filter by selecting a tag', () => {
+            getActiveTags()
+                .first()
+                .click()
+                .its('length')
+                .should('gt', 0);
+        });
+
+        it('should clear all filters when selecting the "clear filters" button', () => {
+            getClearButton()
+                .click()
+                .wait(500);
```

Comment:
Is this waiting for the same reason as Line 23? Is there a way to centralize what's happening here so it can be changed all at once when we have the solution?

How do you feel about extending third party libraries with more functions?